src/index.ts

```diff
@@ -417,9 +424,19 @@ export class Bigtable {
   appProfileId?: string;
   projectName: string;
   shouldReplaceProjectIdToken: boolean;
+  clientUid = crypto.randomUUID();
   static AppProfile: AppProfile;
   static Instance: Instance;
   static Cluster: Cluster;
+  // Each time a metrics handler is created it introduces significant latency.
+  // Therefore, metrics handlers should be created at the client level and
+  // reused throughout the library to reduce latency:
+  metricsHandlers: IMetricsHandler[];
+  // collectMetrics is a member variable that is used to ensure that if the
+  // user provides a `false` value and opts out of metrics collection that
+  // the metrics collector is ignored altogether to reduce latency in the
+  // client.
+  collectMetrics: boolean;
```

Comment:
nit: Using a verb for a flag can be a bit confusing, it sounds more like a function name. I'd suggest something like `metricsEnabled`